### DISCUSSION TITLE
What is the purpose of the electricity adjustment logic when processing MRIO data? Is this solely an artifact of Exiobase?

### DISCUSSION THREAD

**Discussion by cclin130** [USEPA/USEEIO#41] [Q&A] [answered]
Hello! I don't fully follow why we make an adjustment to electricity values here:

https://github.com/USEPA/USEEIO/blob/master/import_factors_exio/generate_import_factors.py#L81

And similarly, why we do this for `US` values. Perhaps it's because I'm not sure what `e_d['Output']` vs. `e_d[export_field]` are. Thanks in advance for the explanation.

**Comment by bl-young** [comment, marked as answer]
The general approach is that we want to aggregate MRIO sectors when they are more granular than USEEIO using the best available data. Our preference is to use bilateral trade data to the US in the MRIO itself. In this case for EXIOBASE, we determined that the bilateral trade data for the electricity sectors to the US was not well characterized and was leading to strange results (where it would be heavily weighed towards an unexpected fuel, like nuclear). Of course keeping in mind that very few countries export electricity to the US in the first place (see the note below). As a result, instead of using the bilateral trade data as the source of the aggregation, for this sector we use industry output instead as a means to weight across electricity fuel types. (by replacing the values in the "Exports to US" column with those from the "Output" column.

Similarly, we keep the U.S. data from EXIOBASE as a form of validation (so we can compare EXIOBASE estimates of US goods to those in USEEIO). But here again, there are no exports to the US for the US, so we have to sub in industry output data as the means for aggregating/weighting across sectors.



_As a side note:_ 
on the other side of the equation where we assess imports by region to the US, the data for electricity is also limited. As we mention in the [report](https://cfpub.epa.gov/si/si_public_record_Report.cfm?dirEntryId=362470&Lab=CESER), neither Census nor BEA report electricity as an imported good or service in their trade data so we need to pull in the regional accounting of electricity imports by country from EIA. This is done in 
https://github.com/USEPA/USEEIO/blob/6f810a3c6bc93598744a8edfcdf313d95db4e432/import_factors_exio/generate_import_factors.py#L60-L61

**Reply by cclin130** [reply]
Thanks for the reply!

> Similarly, we keep the U.S. data from EXIOBASE as a form of validation (so we can compare EXIOBASE estimates of US goods to those in USEEIO). But here again, there are no exports to the US for the US, so we have to sub in industry output data as the means for aggregating/weighting across sectors.

Could you point me to where in the code the validation occurs? My reading of the code is that we replace EXIOBASE's zero value for US-to-US trade with EXIOBASE's non-zero value of industry US output. This value is then used to incorporate US output as one of the contributors to the calculated import EF.

**Reply by bl-young** [reply]
Sorry - by validation, I mean that we generate a figure in our report that shows the EXIOBASE derived US factors compared to our revised US factors (or maybe USEEIO domestic factors, I forget). We don't directly validate in the code. We just keep the US results in the outputs.

**Reply by cclin130** [reply]
What do you mean by "revised US factors"?

Separately, I want to confirm that I'm following the logic in the report and the code correctly:
1. The `M_m` values do _not_ use data about US production. If a sector had 50% of its share of production made domestically in the US, 25% imported from Canada, 25% from Mexico, `M_m` for that sector would be an average between Canada and Mexico.
2. How this is implemented in the code, despite the fact that you override the bilateral trade data to say that the US is an exporter to itself, is that the import share for the US in the `import_shares.csv` file is not specified, so when we compute the country-weighted average of `M_m`, we do not factor in the US EF for that sector into the import EF.
3. The usefulness in overriding bilateral trade data to say that the US is an exporter to itself is that you can get the US Exiobase sectors mapped to the USEEIO sectors to compare them to the original USEEIO data?

Thanks in advance!

**Reply by bl-young** [reply]
Ahh, perhaps its not a set of figures we included in the report, but on the [Figshare](https://figshare.com/articles/figure/Supporting_Figures_for_Estimating_embodied_environmental_flows_in_international_imports_for_the_USEEIO_Model_/26400619?file=48189175) you'll find some figures that compare the domestic factors in USEEIO with those reported in EXIOBASE.

1. Correct
2. The import_shares file is based on the BEA and Census data, not on trade data in the MRIO. Trade data in the MRIO (bilateral trade in most cases) is used only to aggregate sectors _within_ the MRIO when it is a many:to:one mapping to USEEIO. There are no imports to the US within the import_shares file, so US EFs from EXIOBASE are not included in `M_m`.
3. Essentially yes. Just to use the same approach to develop an EXIOBASE derived EF that aligns with the USEEIO schema.

**Reply by cclin130** [reply]
stellar, this all makes sense, thank you :)
